File: backend/src/main/java/com/gdje_izlazimo/project/dto/request/update/UpdateEventRequest.java
package com.gdje_izlazimo.project.dto.request.update;

import com.gdje_izlazimo.project.dto.request.create.EventTicketTypeRequest;
import com.gdje_izlazimo.project.enums.EventType;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.Size;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record UpdateEventRequest(

        UUID venueId,

        @Size(max = 150, message = "Event name must not exceed 150 characters")
        String name,

        String description,

        @Future(message = "Event date/time must be in the future")
        LocalDateTime eventDateTime,

        LocalDateTime eventEndDateTime,

        String locationName,
        String locationAddress,

        EventType eventType,

        String externalOrganizerName,
        String externalOrganizerInstagram,

        Boolean featured,

        Double latitude,
        Double longitude,

        @Valid
        List<EventTicketTypeRequest> ticketTypes
) {}

File: frontend/src/app/core/models/events/update-event.request.ts
import { EventTicketTypeRequest } from './create-event.request';

export interface UpdateEventDto {
  name?: string;
  description?: string;
  eventDateTime?: string;
  eventEndDateTime?: string | null;
  locationName?: string | null;
  locationAddress?: string | null;
  eventType?: string | null;
  externalOrganizerName?: string | null;
  externalOrganizerInstagram?: string | null;
  featured?: boolean;
  imageUrl?: string;
  ticketTypes?: EventTicketTypeRequest[];
  latitude?: number | null;
  longitude?: number | null;
}